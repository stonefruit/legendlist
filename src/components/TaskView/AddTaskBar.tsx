// TODO: Add state testing
import { useState } from 'react'
import { PlusIcon } from '@heroicons/react/solid'
import { classNames } from '../../utils'

type Props = {
  addTask(task: { name: string }): void
}
function AddTaskBar({ addTask }: Props) {
  const [name, setName] = useState('')

  // FUNCTIONS

  const validateAndAddTask = () => {
    if (name) {
      addTask({ name })
      setName('')
    }
  }

  const onClickAddTask = () => {
    validateAndAddTask()
  }

  const onChangeContent = (e: React.ChangeEvent<HTMLInputElement>) => {
    setName(e.target.value)
  }

  const onKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      validateAndAddTask()
    }
  }

  return (
    <div className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg">
      <div className="flex-1 px-4 flex justify-between">
        <div className="flex-1 flex">
          <div className="relative w-full text-gray-400 focus-within:text-gray-300">
            <div
              id="add-button"
              onClick={onClickAddTask}
              className={classNames(
                name && 'cursor-pointer',
                'absolute inset-y-0 left-0 flex items-center'
              )}
            >
              <PlusIcon
                className={classNames(
                  name && 'text-yellow-500 hover:text-yellow-600',
                  'h-4 w-4'
                )}
                aria-hidden="true"
              />
            </div>
            <input
              value={name}
              onChange={onChangeContent}
              id="new-task-text-field"
              className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
              placeholder="Add Task. Press Enter to save."
              type="addTask"
              name="addTask"
              onKeyUp={onKeyUp}
            />
          </div>
        </div>
      </div>
    </div>
  )
}

export default AddTaskBar
