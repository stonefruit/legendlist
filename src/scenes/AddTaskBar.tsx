import { PlusIcon } from '@heroicons/react/solid'
import { useState } from 'react'
import { classNames } from '../utils'

type Props = {
  addTask(task: { name: string }): Promise<void>
}
function AddTaskBar({ addTask }: Props) {
  const [name, setName] = useState('')

  const validateAndAddTask = async () => {
    if (name) {
      await addTask({ name })
      setName('')
    }
  }

  const onClickAddTask = async () => {
    await validateAndAddTask()
  }

  const onChangeContent = (e: React.ChangeEvent<HTMLInputElement>) => {
    setName(e.target.value)
  }

  const onKeyUp = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      await validateAndAddTask()
    }
  }

  return (
    <div className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg">
      <div className="flex-1 px-4 flex justify-between">
        <div className="flex-1 flex">
          <div className="relative w-full text-gray-400 focus-within:text-gray-300">
            <div
              onClick={onClickAddTask}
              className={classNames(
                name && 'cursor-pointer',
                'absolute inset-y-0 left-0 flex items-center'
              )}
            >
              <PlusIcon
                className={classNames(
                  name && 'text-yellow-500 hover:text-yellow-600',
                  'h-4 w-4'
                )}
                aria-hidden="true"
              />
            </div>
            <input
              value={name}
              onChange={onChangeContent}
              id="search-field"
              className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
              placeholder="Add Task. Press Enter to save."
              type="addTask"
              name="addTask"
              onKeyUp={onKeyUp}
            />
          </div>
        </div>
      </div>
    </div>
  )
}

export default AddTaskBar
