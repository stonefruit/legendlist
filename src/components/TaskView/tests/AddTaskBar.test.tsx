import TestRenderer from 'react-test-renderer'
import AddTaskBar from '../AddTaskBar'

const { act } = TestRenderer

test('AddTaskBar', async () => {
  const component = TestRenderer.create(<AddTaskBar addTask={async () => {}} />)
  const testInstance = component.root
  let tree = component.toJSON() as any
  expect(tree).toMatchSnapshot()

  const addButton = testInstance.findByProps({ id: 'add-button' })
  await addButton.props.onClick()
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  const addTaskTextField = testInstance.findByProps({
    id: 'new-task-text-field',
  })
  act(() => {
    addTaskTextField.props.onChange({ target: { value: 'TEST' } })
  })
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  await act(async () => {
    await addTaskTextField.props.onKeyUp({ key: 'Enter' })
  })
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  const addTaskTextField2 = testInstance.findByProps({
    id: 'new-task-text-field',
  })
  expect(addTaskTextField2.props.value).toBe('')
})
