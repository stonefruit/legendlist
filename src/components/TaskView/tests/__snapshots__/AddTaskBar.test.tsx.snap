// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskBar 1`] = `
<div
  className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg"
>
  <div
    className="flex-1 px-4 flex justify-between"
  >
    <div
      className="flex-1 flex"
    >
      <div
        className="relative w-full text-gray-400 focus-within:text-gray-300"
      >
        <div
          className="absolute inset-y-0 left-0 flex items-center"
          id="add-button"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="h-4 w-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
          id="new-task-text-field"
          name="addTask"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Add Task. Press Enter to save."
          type="addTask"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AddTaskBar 2`] = `
<div
  className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg"
>
  <div
    className="flex-1 px-4 flex justify-between"
  >
    <div
      className="flex-1 flex"
    >
      <div
        className="relative w-full text-gray-400 focus-within:text-gray-300"
      >
        <div
          className="absolute inset-y-0 left-0 flex items-center"
          id="add-button"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="h-4 w-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
          id="new-task-text-field"
          name="addTask"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Add Task. Press Enter to save."
          type="addTask"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AddTaskBar 3`] = `
<div
  className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg"
>
  <div
    className="flex-1 px-4 flex justify-between"
  >
    <div
      className="flex-1 flex"
    >
      <div
        className="relative w-full text-gray-400 focus-within:text-gray-300"
      >
        <div
          className="cursor-pointer absolute inset-y-0 left-0 flex items-center"
          id="add-button"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="text-yellow-500 hover:text-yellow-600 h-4 w-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
          id="new-task-text-field"
          name="addTask"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Add Task. Press Enter to save."
          type="addTask"
          value="TEST"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AddTaskBar 4`] = `
<div
  className="flex mt-2 h-9 bg-gray-100 border border-opacity-0 focus-within:border-opacity-100 border-gray-200 focus-within:border-yellow-500 mx-5 rounded-lg"
>
  <div
    className="flex-1 px-4 flex justify-between"
  >
    <div
      className="flex-1 flex"
    >
      <div
        className="relative w-full text-gray-400 focus-within:text-gray-300"
      >
        <div
          className="absolute inset-y-0 left-0 flex items-center"
          id="add-button"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="h-4 w-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          className="text-sm block bg-gray-100 w-full h-full pl-6 pr-3 py-2 border-transparent focus:text-gray-900 placeholder-gray-400 focus:outline-none focus:placeholder-gray-300 focus:ring-0 focus:border-transparent"
          id="new-task-text-field"
          name="addTask"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Add Task. Press Enter to save."
          type="addTask"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

